pex/interpreter.py

```diff
@@ -308,8 +308,20 @@ class PythonInterpreter(object):
     _PYTHON_INTERPRETER_BY_NORMALIZED_PATH = {}  # type: Dict
 
     @staticmethod
-    def _normalize_path(path):
-        return os.path.realpath(path)
+    def canonicalize_path(path):
```

Comment:
The stdlib (rightly I think) refers to what `os.path.realpath` does as "canonicalizing", so it might be clearer to name this method something else... maybe just `abspath` or `absolute_path`...?

Reply:
Well, the point of the name as the docs clarify is that this is canonical for PythonInterpreter irrespective of what other libs may think. The intent is opaque canonicalization and mentioning abs defeats that. I could switch back to normalize language, but that also runs up against os.path.normpath expectations so would seem no more acceptable given your logic. Can you think of a word you like not used by the stdlib that keeps this opaque as to impl but clear as to intent?